source/grid/grid_generator_pipe_junction.cc

```diff
@@ -69,6 +69,194 @@ namespace
             ((y > 0) ? data.cotangent_azimuth_half_right :
                        data.cotangent_azimuth_half_left);
     }
+
+
+
+    /**
+     * Pipe segment manifold description.
+     *
+     * The manifold class is too specific to being of any other use than for the
+     * pipe junction geometry.
+     */
+    template <int dim, int spacedim = dim>
+    class Manifold : public ChartManifold<dim, spacedim, 3>
+    {
+    public:
+      /**
+       * Constructor. The manifold described is a pipe segment whose central
+       * axis points in @p direction and goes through the given
+       * @p point_on_axis.
+       *
+       * The vector @p normal_direction needs to be perpendicular to
+       * @p direction and acts as the reference for azimuth angles. In other
+       * words, $\phi = 0$ corresponds to @p normal_direction. For the pipe
+       * junction geometry, it needs to point in the same direction as the
+       * projection of the bifurcation edge on the opening plane, as this is
+       * the reference direction for azimuth angles by construction.
+       *
+       * Both @p normal_direction and @p direction have to be unit vectors.
+       *
+       * We need @p data to map the pipe segment to a regular cylinder and back.
+       *
+       * The @p tolerance value is used to validate both direction vectors and
+       * to determine if a point in on the axis.
+       */
+      Manifold(const Tensor<1, spacedim> &normal_direction,
+               const Tensor<1, spacedim> &direction,
+               const Point<spacedim> &    point_on_axis,
+               const AdditionalData &     data,
+               const double               tolerance = 1e-10);
+
+      /**
+       * Make a clone of this Manifold object.
+       */
+      virtual std::unique_ptr<dealii::Manifold<dim, spacedim>>
+      clone() const override;
+
+      /**
+       * Compute the cylindrical coordinates $(r, \phi, \lambda)$ for the given
+       * space point and map them to a cylinder of height one, where $r$ denotes
+       * the distance from the axis, $\phi$ the angle between the given point
+       * and the normal direction, and $\lambda$ the axial position.
+       */
+      virtual Point<3>
+      pull_back(const Point<spacedim> &space_point) const override;
+
+      /**
+       * Compute the Cartesian coordinates for a chart point given in
+       * cylindrical coordinates $(r, \phi, \lambda)$ on a cylinder of height
+       * one, where $r$ denotes the distance from the axis, $\phi$ the angle
+       * between the given point and the normal direction, and $\lambda$ the
+       * axial position.
+       */
+      virtual Point<spacedim>
+      push_forward(const Point<3> &chart_point) const override;
+
+    protected:
```

Comment:
What is the reason why these members could not be `private` as well?

Reply:
Honestly, I copied the class's structure from `CylindricalManifold`, which has them `protected` as well. I don't know why they were protected in the first class.

Reply:
See https://github.com/dealii/dealii/pull/13307.